### ISSUE TITLE
Seems you are missing convertToDatabaseValue

### ISSUE THREAD

**Issue by basz** [VasekPurchart/Doctrine-Date-Time-Immutable-Types#7]
without adding the following an exception is still thrown

```
// src/DateImmutableType.php

    /**
     * {@inheritdoc}
     */
    public function convertToDatabaseValue($value, AbstractPlatform $platform)
    {
        if (null === $value) {
            return $value;
        }

        if ($value instanceof DateTimeImmutable) {
            return $value->format($platform->getDateTimeFormatString());
        }

        throw ConversionException::conversionFailedInvalidType($value, $this->getName(), ['null', 'DateTimeImmutable']);
    }
```

**Comment by basz** [comment]
ps happy to PR if accepted

**Comment by basz** [comment]
Well, it's missing for now the upcoming 2.6 (dev-master) which seems to be less forgiving ...

**Comment by VasekPurchart** [comment]
Hi, thanks for letting me know about this, I searched when this got changed and the originating PR seems to be https://github.com/doctrine/dbal/pull/869.

I have always tried to keep the code in the immutable types as close as possible to the original one (to provide easiest migration path), but with this change I am not sure how to approach it. The problem is the new validation logic is using previously not existing method `ConversionException::conversionFailedInvalidType`, which means if the logic is going to change, then also the minimum requirement on DBAL has to be changed. Which I think is currently not a good idea since it is not a stable release. Other option would be testing the `ConversionException` in runtime, whether the method exists, which I don't like, because this is 1) not really pretty, 2) drifts the code apart from the original version.

Other option, which comes to my mind is proposing in DBAL, that they change the check to DateTimeInterface, because that is what is really needed by the implementation, but I am not sure If they would accept it. Although introducing that check is essentially a BC break, because there was no documented type before. I would like this solution best, because even after this lands in (presumably) 2.6 it would force the users of this library to choose if they use a "before 2.6" version, or an "after 2.6" version :(.

**Comment by basz** [comment]
I agree.

It wasn't after I posted that I found out that dev-master does it different. 

I do not see what's the best way to proceed for you. You could create a second branch (do you have releases?), or perhaps look into the 'conflicts' property of composer.json 
I don't think doctrine wants to introduce DateTimeInterface as that would increase the dependency on php to 55 and I think they don't want do do that until 3.0. They might as well introduce those helpers themselves if they would...

For me the issue is resolved as I have persuaded the maintainer of a lib that used DateTimeImmutable to use DateTime in his entities. that works too ;)

Thanks for the effort!

**Comment by VasekPurchart** [comment]
Compatibility with 2.6.x released as [0.4](https://github.com/VasekPurchart/Doctrine-Date-Time-Immutable-Types/releases/tag/0.4).
